lesson26/src/js/login.js

```diff
@@ -1,12 +1,60 @@
 import * as validation from "./validation";
 import { togglePassword } from "./module/toggle-password";
+import { Chance } from "chance";
+const chance = new Chance();
 
 const submitButton  = document.querySelector('[data-id="submit_button"]');
+const userNameInput = document.getElementById('username');
+const passwordInput = document.getElementById('current-password');
+const togglePasswordButton = document.getElementById('js-toggle-password');
 
 submitButton.addEventListener("click", (e) => {
     e.preventDefault();
+    init();
 });
 
+const isRegisteredUser = (inputValues) => {
+    const userData = {
+        name:"Nakagawa",
+        email: "nakagawa@sample.com",
+        password: "N302aoe3"
+    }
+
+    return ( inputValues.userName === userData.name || userData.email ) && inputValues.password === userData.password;
+}
+
+const checkRegisteredUser = () => {
+    const inputValues = {
+        userName : userNameInput.value,
+        password : passwordInput.value
+    }
+
+    return new Promise((resolve,reject) => {
+        if( isRegisteredUser(inputValues) ){
+            resolve({ token: chance.guid(), ok: true , code: 200 })
+        }else{
+            reject({ ok: false, code: 401 })
+        }
+    }) 
+}
+
+const awaitCheckRegisteredUser = async () => {
+    try{
+        const response = await checkRegisteredUser();
+        if(!response) return;
```

Comment:
Q.
Is an early return required here? 👀

Reply:
Sorry for my ignorance.
Is there a problem with this?

I am checking with a negative statement here because I want to check for the presence of a RESPONSE. 
It is written in negative statement, so it is returned early to avoid executing later processing.

Reply:
日本語ですみません:bow:
tryの中で何かしらのエラーが起きた場合、catch節内の処理が行われるはずなので、ここの早期リターンは必要ないのではと思いました。また、tryで問題が起こった際の処理がcatch節だと思うので、tryの中でプロセスを止める処理をするのが違和感がありました。今回の「値がない場合」をチェックしたいのであれば、エラーをスローしてcatchでエラーをキャッチするとかのがいいのかなと思いました。

が、すみません、自分も自信なく...:sob:
一旦コメントさせていただきます:bow:

Reply:
詳しくご説明ありがとうございます:pray:
確かにおっしゃる通りです。
tryで発生した例外は、catch節で処理される（responceが存在しない場合もキャッチする）ので、
ここでの早期リターンはいらないのだと理解できました。

何度も申し訳ありません。修正しましたので、ご確認頂けると幸いです！
https://github.com/nakagawamai/js-lesson/pull/31/commits/ad50586315324a95084f7f3cd5c23cf492cb3e76